torchvision/prototype/datapoints/_dataset_wrapper.py

```diff
@@ -0,0 +1,411 @@
+
+    spatial_size = tuple(F.get_spatial_size(image))
+    batched_target["boxes"] = datapoints.BoundingBox(
+        batched_target["bbox"],
+        format=datapoints.BoundingBoxFormat.XYXY,
+        spatial_size=spatial_size,
+    )
+    batched_target["masks"] = datapoints.Mask(
+        torch.stack(
+            [
+                segmentation_to_mask(segmentation, spatial_size=spatial_size)
+                for segmentation in batched_target["segmentation"]
+            ]
+        ),
+    )
+    batched_target["labels"] = datapoints.Label(
+        batched_target["category_id"], categories=get_coco_detection_categories(dataset)
+    )
+
+    return image, batched_target
+
+
+_WRAPPERS[datasets.CocoDetection] = coco_dectection_wrapper
+_WRAPPERS[datasets.CocoCaptions] = identity
+
+
+VOC_DETECTION_CATEGORIES = [
+    "__background__",
+    "aeroplane",
+    "bicycle",
+    "bird",
+    "boat",
+    "bottle",
+    "bus",
+    "car",
+    "cat",
+    "chair",
+    "cow",
+    "diningtable",
+    "dog",
+    "horse",
+    "motorbike",
+    "person",
+    "pottedplant",
+    "sheep",
+    "sofa",
+    "train",
+    "tvmonitor",
+]
+VOC_DETECTION_CATEGORY_TO_IDX = dict(zip(VOC_DETECTION_CATEGORIES, range(len(VOC_DETECTION_CATEGORIES))))
+
+
+def voc_detection_wrapper(dataset, sample):
+    image, target = sample
+
+    batched_instances = list_of_dicts_to_dict_of_lists(target["annotation"]["object"])
+
+    target["boxes"] = datapoints.BoundingBox(
+        [[int(bndbox[part]) for part in ("xmin", "ymin", "xmax", "ymax")] for bndbox in batched_instances["bndbox"]],
+        format=datapoints.BoundingBoxFormat.XYXY,
+        spatial_size=(image.height, image.width),
+    )
+    target["labels"] = datapoints.Label(
+        [VOC_DETECTION_CATEGORY_TO_IDX[category] for category in batched_instances["name"]],
+        categories=VOC_DETECTION_CATEGORIES,
+    )
+
+    return image, target
+
+
+_WRAPPERS[datasets.VOCDetection] = voc_detection_wrapper
+
+
+def sbd_wrapper(dataset, sample):
+    if dataset.mode == "boundaries":
+        raise RuntimeError(
+            "SBDataset with mode='boundaries' is currently not supported by this wrapper. "
+            "If this would be helpful for you, please open an issue at https://github.com/pytorch/vision/issues."
+        )
+
+    image, target = sample
+    return image, datapoints.Mask(F.to_image_tensor(target).squeeze(0))
+
+
+_WRAPPERS[datasets.SBDataset] = sbd_wrapper
+
+
+def celeba_wrapper(dataset, sample):
+    image, target = sample
+    return wrap_target_by_type(
+        dataset,
+        target,
+        {
+            "identity": datapoints.Label,
+            "bbox": lambda item: datapoints.BoundingBox(
+                item, format=datapoints.BoundingBoxFormat.XYWH, spatial_size=(image.height, image.width)
+            ),
+        },
+        # FIXME: Failing on "attr" here is problematic, since it is the default
+        fail_on=["attr", "landmarks"],
```

Comment:
Without `GenericDatapoint` or a different handling as explored in https://github.com/pytorch/vision/pull/6663#discussion_r1093087483, these two would be treated as images since they are tensors. 

As the comment states, `"attr"` is the default, so we shouldn't fail here if possible. I'm not sure how exactly `CelebA` is used in the wild, but one option could be to wrap `attr` in `datapoints.OneHotLabel`. That would be only somewhat true, since `attr` can have multiple `1`'s depending on which facial features are present. That being said, we had some intense offline discussions about the `OneHotLabel` name before and weren't able to come up with a better one. The new `MixUp` and `CutMix` transforms also violate the "One" part of the name, since they blend labels and so only `sum(one_hot_label) == 1`.

`"landmarks"` are actually keypoints that we should support at some point in the future, given that we have models for that. There is #5326 for that, which had some positive feedback, but was deemed low priority and thus not followed through.